dask/dataframe/io/parquet.py

```diff
@@ -174,11 +147,147 @@ def _read_parquet_row_group(open, fn, index, columns, rg, series, categories,
         return df
 
 
+# ----------------------------------------------------------------------
+# PyArrow interface
+
+def _read_pyarrow(fs, paths, file_opener, columns=None, filters=None,
+                  categories=None, index=None):
+    if not pyarrow_parquet:
+        raise ImportError("pyarrow.parquet not installed")
+
+    api = pyarrow_parquet
+
+    if filters is not None:
+        raise NotImplemented("Predicate pushdown not implemented")
+
+    if categories is not None:
+        raise NotImplemented("Categorical reads not yet implemented")
+
+    if isinstance(columns, tuple):
+        columns = list(columns)
+
+    columns = columns
+
+    dataset = api.ParquetDataset(paths)
+    schema = dataset.schema.to_arrow_schema()
+    task_name = 'read-parquet-' + tokenize(dataset, columns)
+
+    pieces = dataset.pieces
+
+    if columns is None:
+        all_columns = schema.names
+    else:
+        all_columns = columns
+
+    if not isinstance(all_columns, list):
+        out_type = Series
+        all_columns = [all_columns]
+    else:
+        out_type = DataFrame
+
+    divisions = (None,) * (len(dataset.pieces) + 1)
```

Comment:
Yep. We should really hustle to do this ASAP so we can do the work in both fastparquet and pyarrow to implement the standard index metadata